src/tools/unpack.c

```diff
@@ -937,32 +939,38 @@ void unpack_cleanup(void *user_data)
     for (int i=0; i<opts->n_sourcedirs; i++) {
         FREE_VAR(opts->sourcedirs[i]);
     }
+
+    sfmf_control_close();
 }
 
 static void draw_progress(struct UnpackOptions *opts, int i, const char *message)
 {
     static float last_progress = -1.f;
 
-    if (opts->progress) {
-        float partial = (opts->header.entries_length && i >= 0)
-            ? fminf(1.f, (float)(i) / (float)(opts->header.entries_length))
-            : (i == -1 ? 0.f : 1.f);
+    float partial = (opts->header.entries_length && i >= 0)
+        ? fminf(1.f, (float)(i) / (float)(opts->header.entries_length))
+        : (i == -1 ? 0.f : 1.f);
 
-        float progress = fminf(1.f, ((float)(opts->steps.current) + partial) / (float)(opts->steps.total));
+    float progress = fminf(1.f, ((float)(opts->steps.current) + partial) / (float)(opts->steps.total));
 
-        if (i >= 0 && progress - last_progress < 0.005f) {
-            // Avoid excessive status updates
-            return;
-        }
+    if (i >= 0 && progress - last_progress < 0.005f) {
```

Comment:
Just wondering: Does progress ever go down (several progress animations or something)? i.e. does last_progress need to be cleared at some conditions?